src/components/List/Container.js

```diff
@@ -108,7 +97,14 @@ class ListContainer extends Component {
 
     return (
       <Container onClick={this.onExpandOrCollapse} {...rest}>
-        <ContainerProvider value={this.state}>
+        <ContainerProvider
```

Comment:
Are we sure we want to pass a brand new object to the value prop of ContainerProvider? I'm concerned this will cause unnecessary rerenders. Perhaps we should extract this object out.

```
const someState = { ... }
<ContainerProvider value={someState}>
```